tests/unit/utils/test_git_shell.py

```diff
@@ -375,3 +376,31 @@ def test_tar_from_ref_and_filepaths(tmp_path):
         filenames = tar.getnames()
         # THEN only first file in in tar
         assert filenames == [first_file_name]
+
+
+def test_git_ls_unstaged(tmp_path):
+    # GIVEN a repository
+    repo = Repository.create(tmp_path)
+    repo.create_commit()
+
+    # AND a submodule
+    submodule = Repository.create(repo.path / "submodule")
+    submodule.create_commit()
+
+    repo.git("submodule", "add", submodule.path)
+
+    # AND unstaged files
+    repo_file = repo.path / "repo_file"
+    submodule_file = repo.path / "submodule" / "submodule_file"
+    repo_file.write_text("Content")
+    submodule_file.write_text("Content")
+
+    # WHEN listing unstaged files
+    unstaged_files = git_ls_unstaged(wd=repo.path)
+
+    # THEN unstaged files in repo and submodule are returned
+    # as relative OR absolute paths
+    assert all(
+        any(item in unstaged_files for item in [str(path), path.name])
+        for path in [repo_file, submodule_file]
+    ), "Not all paths are found in unstaged_files"
```

Comment:
This got me trying `git_ls_unstaged()` manually and I think it would be more consistent if it had the same behavior as `git_ls()`: all returned paths should be relative to the directory passed as its argument.

I ask this because I am worried this could bite us back in the future.

Reply:
There was a bit of path reconstruction to do, but I think we're good now!